pkg/virt-controller/watch/vm.go

```diff
@@ -527,6 +527,7 @@ func (c *VMController) setupVMIFromVM(vm *virtv1.VirtualMachine) *virtv1.Virtual
 	vmi.ObjectMeta = vm.Spec.Template.ObjectMeta
 	vmi.ObjectMeta.Name = basename
 	vmi.ObjectMeta.GenerateName = basename
+	vmi.ObjectMeta.Namespace = vm.ObjectMeta.Namespace
```

Comment:
it's fine to set this I suppose. it shouldn't make any difference though. We create the vmi in the same namespace as its parent vm, so namespace gets assigned regardless

Reply:
right, eventually the VMI would be created within the same namespace - I didn't investigate the whole flow but without setting it here, the mutating-hook receives the VMI without the namespace and so the relevant namespace-limits are not applied.

Reply:
right, eventually the VMI would be created within the same namespace - I didn't investigate the whole flow but this is probably set relatively late since without setting it here, the mutating-hook receives the VMI without the namespace and so the relevant namespace-limits are not applied.